invokeai/backend/stable_diffusion/extensions/lora_patcher.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+from contextlib import contextmanager
+from typing import TYPE_CHECKING, Dict, Iterator, List, Optional, Tuple
+
+import torch
+from diffusers import UNet2DConditionModel
+
+from invokeai.backend.stable_diffusion.extensions.base import ExtensionBase
+from invokeai.backend.util.devices import TorchDevice
+
+if TYPE_CHECKING:
+    from invokeai.app.invocations.model import LoRAField
+    from invokeai.app.services.shared.invocation_context import InvocationContext
+    from invokeai.backend.lora import LoRAModelRaw
+
+
+class LoRAPatcherExt(ExtensionBase):
+    def __init__(
+        self,
+        node_context: InvocationContext,
+        loras: List[LoRAField],
+        prefix: str,
+    ):
+        super().__init__()
+        self._loras = loras
+        self._prefix = prefix
+        self._node_context = node_context
+
+    @contextmanager
+    def patch_unet(self, unet: UNet2DConditionModel, cached_weights: Optional[Dict[str, torch.Tensor]] = None):
+        def _lora_loader() -> Iterator[Tuple[LoRAModelRaw, float]]:
+            for lora in self._loras:
+                lora_info = self._node_context.models.load(lora.lora)
+                lora_model = lora_info.model
+                yield (lora_model, lora.weight)
+                del lora_info
+            return
+
+        yield self._patch_model(
+            model=unet,
+            prefix=self._prefix,
+            loras=_lora_loader(),
+            cached_weights=cached_weights,
+        )
+
+    @classmethod
+    @contextmanager
+    def static_patch_model(
+        cls,
+        model: torch.nn.Module,
+        prefix: str,
+        loras: Iterator[Tuple[LoRAModelRaw, float]],
+        cached_weights: Optional[Dict[str, torch.Tensor]] = None,
+    ):
+        modified_cached_weights, modified_weights = cls._patch_model(
+            model=model,
+            prefix=prefix,
+            loras=loras,
+            cached_weights=cached_weights,
+        )
+        try:
+            yield
+
+        finally:
+            with torch.no_grad():
+                for param_key in modified_cached_weights:
+                    model.get_parameter(param_key).copy_(cached_weights[param_key])
+                for param_key, weight in modified_weights.items():
+                    model.get_parameter(param_key).copy_(weight)
+
+    @classmethod
+    def _patch_model(
```

Comment:
Yes, as I mentioned in merge plan - I planning to replace calls to ModelPatcher with calls to `static_patch_model` in this class.
For now not changed, so that we can easily compare that results are same.